crates/utils/src/utils/validation.rs

```diff
@@ -257,16 +255,23 @@ pub fn build_and_check_regex(regex_str_opt: &Option<&str>) -> LemmyResult<Option
   )
 }
 
-pub fn clean_url_params(url: &Url) -> Url {
-  let mut url_out = url.clone();
-  if let Some(query) = url.query() {
-    let new_query = query
-      .split_inclusive('&')
-      .filter(|q| !CLEAN_URL_PARAMS_REGEX.is_match(q))
-      .collect::<String>();
-    url_out.set_query(Some(&new_query));
+/// Cleans a url of tracking parameters.
+pub fn clean_url(url: &Url) -> LemmyResult<Url> {
+  match URL_CLEANER.clear_single_url(url.as_str()) {
+    Ok(res) => Ok(Url::parse(&res)?),
+    // If there are any errors, just return the original url
+    Err(_) => Ok(url.clone()),
+  }
+}
+
+/// Cleans all the links in a string of tracking parameters.
+pub fn clean_urls_in_text(text: &str) -> String {
+  let finder = LinkFinder::new();
+  match URL_CLEANER.clear_text(text, &finder) {
```

Comment:
@jendrikw Another small note, for us it would be convenient to have a method without the Linkfinder parameter, so we dont have to add that crate as a direct dependency.

Reply:
The library also has a `clear_markdown(` function, where it finds the links not scanning the text, but passing in a single param, the parsed markdown via MarkdownIt. 

I thought about using it since we already do this in the `process_markdown` to rewrite image links. But the Linkfinder seemed better.

All this to say that I don't care which one we do, but we could remove the LinkFinder, and restructure how we parse the markdown a little.

Reply:
The LinkFinder would make sense if we convert plaintext links to markdown links, but it looks like thats not happening now. So `clear_markdown()` makes more sense.

Reply:
Nevermind, I started to work on this, and realized its a bad idea, for at least 2 reasons:

-  The markdown library has no easy way to convert back from a markdown node, to markdown text. You'd have to create something that walks through the entire tree yet again : https://github.com/markdown-it-rust/markdown-it/issues/30
- The `clearurls.clear_markdown` edits the nodes in place, which makes it much less clean to use.

Reply:
I've added a method that doesn't require a LinkFinder in 0.0.3

Reply:
Thx! Updated.